x-pack/plugins/cases/public/containers/configure/use_get_case_configuration.test.tsx

```diff
@@ -24,90 +22,72 @@ describe('Use get case configuration hook', () => {
     jest.clearAllMocks();
   });
 
-  it('calls the api when invoked with the correct parameters', async () => {
+  it('returns a configuration matching the owner', async () => {
     const spy = jest.spyOn(api, 'getCaseConfigure');
+    const targetConfiguration = {
+      id: 'my-new-configuration-3',
+      owner: mockedTestProvidersOwner[0], // used in the AppMockRenderer
+    };
+    spy.mockResolvedValue([
+      // @ts-expect-error: no need to define all properties
+      { id: 'my-new-configuration-1', owner: 'foo' },
+      // @ts-expect-error: no need to define all properties
+      { id: 'my-new-configuration-2', owner: 'bar' },
+      // @ts-expect-error: no need to define all properties
+      targetConfiguration,
+    ]);
 
-    const { waitForNextUpdate } = renderHook(() => useGetCaseConfiguration(), {
+    const { result, waitForNextUpdate } = renderHook(() => useGetCaseConfiguration(), {
       wrapper: appMockRender.AppWrapper,
     });
-    await waitForNextUpdate();
-
-    expect(spy).toHaveBeenCalledWith({
-      owner: ['securitySolution'],
-      signal: expect.any(AbortSignal),
-    });
-  });
-
-  it('shows a toast error when the api return an error', async () => {
-    const addError = jest.fn();
-    (useToasts as jest.Mock).mockReturnValue({ addError });
 
-    const spy = jest.spyOn(api, 'getCaseConfigure').mockRejectedValue(new Error('error'));
-
-    const { waitForNextUpdate } = renderHook(() => useGetCaseConfiguration(), {
-      wrapper: appMockRender.AppWrapper,
-    });
     await waitForNextUpdate();
 
-    await waitFor(() => {
-      expect(spy).toHaveBeenCalledWith({
-        owner: ['securitySolution'],
-        signal: expect.any(AbortSignal),
-      });
-
-      expect(addError).toHaveBeenCalled();
-    });
+    /**
+     * The response after fetching
+     */
+    // @ts-expect-error: data is defined
+    expect(result.all[1].data).toEqual(targetConfiguration);
```

Comment:
nit: better to use `result.current` to be sure that the latest value returned by the hook is what you expected.